lib/link.js

```diff
@@ -127,6 +127,7 @@ Link.prototype.detach = function() {
 Link.prototype.forceDetach = function() {
   debug('Force-detach for ' + this.name + '. Current state: ' + this.state());
   this.linkSM.forceDetach();
+  this._detached();
```

Comment:
I worry slightly here, since we're now emitting a Detached event when forceDetach is called. If re-attach is triggered when that happens, it'd be bad imho, but I don't remember the flow well enough to dissuade this fix. Perhaps keep frame as undefined instead of coalescing to {} below, and only emit the event if it's defined? Also, what does shouldReattach do in this case?